File: VB-Examples/24_Workbook/AddCustomProperties.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace AddCustomProperties
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load a workbook from a specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\AddCustomProperties.xlsx")

            ' Add a custom document property "_MarkAsFinal" with a value of True
            workbook.CustomDocumentProperties.Add("_MarkAsFinal", True)

            ' Add custom document properties with their respective names and values
            workbook.CustomDocumentProperties.Add("The Editor", "E-iceblue")
            workbook.CustomDocumentProperties.Add("Phone number", 81705109)
            workbook.CustomDocumentProperties.Add("Revision number", 7.12)
            workbook.CustomDocumentProperties.Add("Revision date", Date.Now)

            ' Save the modified workbook to a new file with the specified file format (Version2013)
            workbook.SaveToFile("AddCustomProperties_result.xlsx", FileFormat.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("AddCustomProperties_result.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/24_Workbook/AddCustomProperties.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace AddCustomProperties
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\AddCustomProperties.xlsx");

            // Load the document from disk
            workbook.CustomDocumentProperties.Add("_MarkAsFinal", true);
            
            // Add other custom properties to the workbook
            workbook.CustomDocumentProperties.Add("The Editor", "E-iceblue");
            workbook.CustomDocumentProperties.Add("Phone number", 81705109);
            workbook.CustomDocumentProperties.Add("Revision number", 7.12);
            workbook.CustomDocumentProperties.Add("Revision date", DateTime.Now);

            // Save the document
            workbook.SaveToFile("AddCustomProperties_result.xlsx", FileFormat.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("AddCustomProperties_result.xlsx");
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
